### PR TITLE
Add vale linter github action

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: .github/workflows/vale.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+        with:
+          repository: splunk/vale-splunk-style-guide
+          path: vale-splunk-style-guide
+
+      - run: |
+          cat <<EOF > .vale.ini
```
Comment (theletterf) [inline .github/workflows/vale.yml:29]: Is this Vale ini in conflict with the StylesPath below?

File: .github/workflows/vale.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          StylesPath = vale-splunk-style-guide/styles
+          MinAlertLevel = suggestion
+          
+          [*.{ md,rst }]
+          BasedOnStyles = Splunk
+          
+          Vocab = splunk
```
Comment (theletterf) [inline .github/workflows/vale.yml:36]: This one is not needed — we still don't have a vocabulary

**Review by theletterf — CHANGES_REQUESTED** [review]

File: .github/workflows/vale.yml (hunk 1 above)
Comment (atoulme) [reply .github/workflows/vale.yml:29]: Sorry, I don't follow. What are you referring to?

File: .github/workflows/vale.yml (hunk 2 above)
Comment (theletterf) [reply .github/workflows/vale.yml:36]: Guess it does no harm!

File: .github/workflows/vale.yml (hunk 1 above)
Comment (theletterf) [reply .github/workflows/vale.yml:29]: Not sure how the setup works for a pipeline/action, but, essentially, the `.vale.ini` file already contains the path to styles. The path in the file must match whatever path in the local filesystem. Wondering if that's the cause for the pipeline not to work.

Comment (atoulme) [reply .github/workflows/vale.yml:29]: I can try entering a nonsensical path and see if it gets picked up.

**Pull request closed without merging**